File: Problems/0059. (M) Spiral Matrix II/Java/solution.java
class Solution {
  public int[][] generateMatrix(int n) {
    // process edges and shrink

    int[][] ret = new int[n][n];
    int l = 0;
    int h = n-1;
    int c = 1;
    while (l < h) {
      for (int i = l; i < h; i++) {
        ret[l][i] = c;
        c++;
      }
      for (int i = l; i < h; i++) {
        ret[i][h] = c;
        c++;
      }
      for (int i = h; i > l-1; i--) {
        ret[h][i] = c;
        c++;
      }
      for (int i = h-1; i > l; i--) {
        ret[i][l] = c;
        c++;
      }
      l++;
      h--;
    }

    if (n % 2 == 1)
      ret[l][l] = c;

    return ret;
  }
}

File: Problems/0059. (M) Spiral Matrix II/Python3/solution.py
class Solution:
  def generateMatrix(self, n: int) -> List[List[int]]:
    # process edges and shrink
    
    ret = [[n**2]*n for _ in range(n)]
    l, h = 0, n-1
    c = 1
    while l < h:
      for i in range(l, h):
        ret[l][i] = c
        c += 1
      for i in range(l, h):
        ret[i][h] = c
        c += 1
      for i in range(h, l-1, -1):
        ret[h][i] = c
        c += 1
      for i in range(h-1, l, -1):
        ret[i][l] = c
        c += 1
      l += 1
      h -= 1
    
    return ret

